### PR TITLE
Silences Pip Install Messages in Code Executors

### LINKED ISSUE #2075: [Feature Request]: automatically quiet pip install in code executors
### Is your feature request related to a problem? Please describe.

Currently in command line code executors, pip install always yields super verbose output like below.

```
Requirement already satisfied: yfinance in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (0.2.37)
Requirement already satisfied: pandas>=1.3.0 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (2.2.1)
Requirement already satisfied: numpy>=1.16.5 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (1.26.4)
Requirement already satisfied: requests>=2.31 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (2.31.0)
Requirement already satisfied: multitasking>=0.0.7 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (0.0.11)
Requirement already satisfied: lxml>=4.9.1 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (5.1.0)
Requirement already satisfied: appdirs>=1.4.4 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (1.4.4)
Requirement already satisfied: pytz>=2022.5 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (2024.1)
Requirement already satisfied: frozendict>=2.3.4 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (2.4.0)
Requirement already satisfied: peewee>=3.16.2 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (3.17.1)
Requirement already satisfied: beautifulsoup4>=4.11.1 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (4.12.3)
Requirement already satisfied: html5lib>=1.1 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from yfinance) (1.1)
Requirement already satisfied: soupsieve>1.2 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from beautifulsoup4>=4.11.1->yfinance) (2.5)
Requirement already satisfied: six>=1.9 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from html5lib>=1.1->yfinance) (1.16.0)
Requirement already satisfied: webencodings in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from html5lib>=1.1->yfinance) (0.5.1)
Requirement already satisfied: python-dateutil>=2.8.2 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from pandas>=1.3.0->yfinance) (2.8.2)
Requirement already satisfied: tzdata>=2022.7 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from pandas>=1.3.0->yfinance) (2024.1)
Requirement already satisfied: charset-normalizer<4,>=2 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from requests>=2.31->yfinance) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from requests>=2.31->yfinance) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from requests>=2.31->yfinance) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in [/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages](https://file+.vscode-resource.vscode-cdn.net/home/ykw5399/.conda/envs/pyautogen/lib/python3.9/site-packages) (from requests>=2.31->yfinance) (2024.2.2)
```

### Describe the solution you'd like

Automatically convert `pip install` command to add `-qqq` to make it quiet, similar to the `JupyterCodeExecutor`

https://github.com/microsoft/autogen/blob/501f8d22726e687c55052682c20c97ce62f018ac/autogen/coding/jupyter/jupyter_code_executor.py#L143

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by WaelKarkoub**
## Related issue number

Closes #2075 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by ekzhu — CHANGES_REQUESTED** [review]
Can you add unit tests to check for the expected behavior? 
1. input code block contains pip install, package is valid, output is empty
2. input code block contains pip install, package is invalid, etc. random strings, output contains error. 

See the existing unit tests for jupyter code executor for reference.

File: test/coding/test_commandline_code_executor.py [hunk 1]
```diff
@@ -218,3 +218,24 @@ def test_valid_relative_path(cls) -> None:
         assert "test.py" in result.code_file
         assert (temp_dir / "test.py").resolve() == Path(result.code_file).resolve()
         assert (temp_dir / "test.py").exists()
+
+
+@pytest.mark.parametrize("cls", classes_to_test)
+@pytest.mark.parametrize("lang", ["bash", "shell", "sh"])
+def test_silent_pip_install(cls, lang: str) -> None:
+    if sys.platform in ["win32"]:
```
Comment (ekzhu) [inline test/coding/test_commandline_code_executor.py:226]: For skipping windows test we can use @pytest.skipif(... ). Why do we want to skip on windows?

Comment (ekzhu) [reply test/coding/test_commandline_code_executor.py:226]: we can add powershell language right?

Comment (WaelKarkoub) [reply test/coding/test_commandline_code_executor.py]: Yeah, I didn't have a way to test it locally. I added a check if windows, skip unix shells, and vice versa.

**Review by ekzhu — APPROVED** [review]

**Pull request merged**